app/models/ability.rb

```diff
@@ -147,6 +147,8 @@ def initialize(params)
             can :create, Sms::Test
           end
 
+          can :regenerate_sms_auth_code, User
+
```

Comment:
I'm realizing that it would probably make sense for an observer to be able to see and regenerate their own auth code (but only their own). Could you add this please?